### PR TITLE
Feature/dependencies accept remote driver

### PR DESCRIPTION
**Pull request by zhenyu-ms**
## Bug / Requirement Description
`RemoteDriver` can be added to `Test`'s driver dependency graph.

## Solution description
update type checking, can it be tested in gh ci?

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [x] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: testplan/common/utils/table.py [hunk 1]
```diff
@@ -15,9 +18,6 @@ def __init__(self, source, placeholder=None):
 
         self._validate_input()
 
-        if self._source and isinstance(self._source[0], (list, tuple)):
```
Comment (Pyifan) [inline testplan/common/utils/table.py:18]: why remove this logging?

File: testplan/runnable/base.py [hunk 2]
```diff
@@ -573,9 +573,7 @@ def add_remote_service(self, remote_service: RemoteService):
         remote_service.start()
 
     def _stop_remote_services(self):
-
-        for name, rmt_svc in self.remote_services.items():
-            self.logger.info("Stopping Remote Server %s", name)
```
Comment (Pyifan) [inline testplan/runnable/base.py:578]: why remove this logging

File: testplan/runnable/base.py [hunk 3]
```diff
@@ -573,9 +573,7 @@ def add_remote_service(self, remote_service: RemoteService):
...
+        for _, rmt_svc in self.remote_services.items():
```
Comment (yuxuan-ms) [inline testplan/runnable/base.py:576]: self.remote_services.values() ?

File: testplan/runnable/base.py (hunk 2 above)
Comment (zhenyu-ms) [reply testplan/runnable/base.py:578]: rmt_svc will log a message inside its `stop` method, i think it's kinda duplicate here

**Review by Pyifan — APPROVED** [review]

**Pull request merged**